reviews-loader/src/prompt.ts

```diff
@@ -0,0 +1,105 @@
+import { Schema, z } from 'zod';
+import { OpenAI } from 'openai';
+import { SentimentEnum } from 'shared-types';
+
+// Chain-of-Thought
+// Few-Shots
+// Prompt Chaining
+interface Prompt<T> {
+  outputSchema: Schema<T>;
+  messages: OpenAI.ChatCompletionMessageParam[];
+  topP: number;
+}
+
+export const getSentimentRatingPrompt = (review: string): Prompt<{ sentiment: SentimentEnum; rating: number }> => {
+  const systemContent = `
+You are an AI that analyzes reviews. First, identify the sentiment (positive, neutral, negative) and rating from 1-10 of the following review.
+
+Examples:
+I: The system seems interesting, but it's a bit too early to tell how effective it will be. I'm waiting to see more functionalities before I can really recommend it.
+O: { "sentiment": "neutral", rating: 5 }
+  
+I: This platform is a game-changer! Having all my customer reviews in one place with clear insights is fantastic. The sentiment analysis helped me identify areas to improve, and the action items are super helpful. Highly recommend!
+O: { "sentiment": "positive", rating: 9 }
+  
+I: The AI analysis seems a bit hit-or-miss. Some of the insights were spot-on, but others felt off base. Also, the pricing seems a bit high for the current feature set.
+O: { "sentiment": "negative", rating: 3 }
+`;
+
+  const userContent = `
+Review: ${review}
+`;
+  return {
+    outputSchema: z.object({ sentiment: z.nativeEnum(SentimentEnum), rating: z.number() }),
```

Comment:
Define this schema before the function. Then you can also you z.infer to infer the type in the function signature